app/src/sharedTest/java/org/oppia/app/administratorcontrols/AppVersionActivityTest.kt

```diff
@@ -0,0 +1,151 @@
+import androidx.test.espresso.matcher.ViewMatchers.isRoot
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.rule.ActivityTestRule
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.oppia.app.BuildConfig
+import org.oppia.app.R
+import org.oppia.app.administratorcontrols.appversion.AppVersionActivity
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import java.text.SimpleDateFormat
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+class AppVersionActivityTest {
+
+  private lateinit var activityScenario: ActivityScenario<AppVersionActivity>
+  @Inject lateinit var context: Context
+  private lateinit var lastUpdateDate: String
+
+  @get:Rule
+  var activityTestRule: ActivityTestRule<AppVersionActivity> = ActivityTestRule(
+    AppVersionActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Before
+  fun setUp() {
+    Intents.init()
+    setUpTestApplicationComponent()
+    activityScenario = launchAppVersionActivityIntent()
+
+     val lastUpdateDateTime = context.packageManager.getPackageInfo(context.packageName, 0).lastUpdateTime
+     lastUpdateDate = getDateTime(lastUpdateDateTime)!!
+
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerAppVersionActivityTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  @Test
+  fun testAppVersionActivity_loadFragment_displaysAppVersion() {
+    launchAppVersionActivityIntent().use {
+      onView(withText(String.format(context.resources.getString(R.string.app_version_name), BuildConfig.VERSION_NAME))).check(matches(isDisplayed()))
+      onView(withText(String.format(context.resources.getString(R.string.app_last_update_date), lastUpdateDate))).check(matches(isDisplayed()))
+//      onView(withText("App Version ${BuildConfig.VERSION_NAME}")).check(matches(isDisplayed()))
+    }
+  }
+
+ @Test
+  fun testAppVersionActivity_loadFragment_onBackPressed_displaysAdministratorControlsActivity() {
+   ActivityScenario.launch<AdministratorControlsActivity>(launchAdministratorControlsActivityIntent(0)).use {
+     onView(withId(R.id.administrator_controls_list)).perform(scrollToPosition<RecyclerView.ViewHolder>(3))
+     onView(withText(R.string.administrator_controls_app_version)).perform(click())
+     Intents.intended(IntentMatchers.hasComponent(AppVersionActivity::class.java.name))
+     onView(isRoot()).perform(pressBack())
+     onView(withId(R.id.administrator_controls_list)).check(matches(isDisplayed()))
+    }
+  }
+
```

Comment:
Press `Control + Alt + L` - reformatting shortcut to fix the formatting.

Reply:
Done